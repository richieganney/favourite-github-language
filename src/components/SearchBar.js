import React, { Component } from 'react';
import axios from 'axios';
import '../styles.css'
import FavouriteLanguage from './FavouriteLanguage';

class SearchBar extends Component {

    constructor(props) {
        super(props);
        this.state =
        {
        username: "",
        languages: []
        };
        this.onChange = this.onChange.bind(this);
        this.onSubmit = this.onSubmit.bind(this);
        this.getRepoLanguages = this.getRepoLanguages.bind(this);
    }

    onChange(e){
      this.setState({ username: e.target.value });
    }

    getRepoLanguages(response){
        return response.data.map(function(repoObject){
            return repoObject.language
        })
    }
    
    onSubmit(e) {
        e.preventDefault();
        let url = process.env.REACT_APP_API_KEY
        axios
          .get(url + `${this.state.username}/repos`)
          .then(response => {
            this.setState({ username: "", languages: this.getRepoLanguages(response) })
          })
          .catch(error => {
            this.setState({username: "", languages: ["That username is invalid, please try again"]})
            console.log(error.response);
          });
    }

    render() {
        return (
            <div>
                <form id="form" className="container" onSubmit={e => {this.onSubmit(e)}}>
                <input type="text" id="usernameSearchBar" value={this.state.username} onChange={this.onChange}></input>
                <div className="search"></div>
                </form>
                <div>
                <FavouriteLanguage
                  languages={this.state.languages}
                 />
                </div>
            </div>
        );
    }
}

export const getRepoLanguages = (response) => {
    return response.data.map(function(repoObject){
        return repoObject.language
    })
}

export default SearchBar;
