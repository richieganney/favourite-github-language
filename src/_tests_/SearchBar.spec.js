import React from "react";
import SearchBar from "../components/SearchBar"
import { create } from "react-test-renderer";

describe("SearchBar component", () => {
  test("Matches the snapshot", () => {
    const searchBar = create(<SearchBar />);
    expect(searchBar.toJSON()).toMatchSnapshot();
  });

// test('request to github api', async () => {
//   const Octokit = require('@octokit/rest')
//   const nock = require('nock')
//   const octokit = new Octokit()
//   const scope = nock('https://api.github.com')
//     .get('/')
//     .reply(200, {})

//   await octokit.request('/')
//   scope.done()
// })
});
