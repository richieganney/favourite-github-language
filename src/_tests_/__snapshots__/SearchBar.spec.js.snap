// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component Matches the snapshot 1`] = `
<div>
  <form
    id="search_bar"
    onSubmit={[Function]}
  >
    <div
      className="search"
    >
      <input
        className="round"
        id="username_search_bar"
        onChange={[Function]}
        placeholder="Enter a github username"
        type="text"
        value=""
      />
    </div>
    <div />
  </form>
</div>
`;
